projects/step-frontend/src/lib/modules/timeseries/components/dashboard/dashboard.component.ts

```diff
@@ -290,19 +335,35 @@ export class DashboardComponent implements OnInit, OnDestroy {
     const context = this.getContext(compareCharts);
     context
       .onTimeSelectionChange()
-      .pipe(
-        switchMap((newRange) => this.handleSelectionChange(newRange)),
-        takeUntilDestroyed(this._destroyRef),
-      )
-      .subscribe();
-  }
-
-  private handleSelectionChange(range?: TimeRange): Observable<any[]> {
-    return this.refreshAllCharts();
+      .pipe(takeUntilDestroyed(this._destroyRef))
+      .subscribe(() => this.refreshAllCharts(false, true));
   }
 
-  private refreshAllCharts(): Observable<any[]> {
-    return forkJoin(this.dashlets?.map((dashlet) => dashlet.refresh())).pipe(defaultIfEmpty([]));
+  /**
+   * @param refreshRanger. Set true if ranger data has to be updated
+   * @param force. If set to false (default), the refresh will be skipped if there is a refresh in progress.
+   * @private
+   */
+  private refreshAllCharts(refreshRanger = false, force = false): void {
+    if (force || !this.refreshInProgress) {
+      this.refreshSubscription?.unsubscribe();
+      this.refreshInProgress = true;
+      this.refreshSubscription = of(null)
+        .pipe(
+          tap(() => (this.refreshInProgress = true)),
+          switchMap(() => {
+            const dashlets$ = this.dashlets?.map((dashlet) => dashlet.refresh());
+            if (refreshRanger) {
+              dashlets$.push(this.refreshRanger());
+            }
+            return forkJoin(dashlets$).pipe(defaultIfEmpty([]));
+          }),
+          tap(() => (this.refreshInProgress = false)),
+        )
+        .subscribe();
+    } else {
+      // skip refresh
+    }
```

Comment:
Instead of empty else block, I would prefer to check the condition at the beginning and exit. It will reduce the code's block nesting and will make it more readable.

```suggestion
    if (this.refreshInProgress && !force) {
        // skip refresh
        return;
    }

    this.refreshSubscription?.unsubscribe();
    this.refreshInProgress = true;
    this.refreshSubscription = of(null)
      .pipe(
        tap(() => (this.refreshInProgress = true)),
        switchMap(() => {
          const dashlets$ = this.dashlets?.map((dashlet) => dashlet.refresh());
          if (refreshRanger) {
            dashlets$.push(this.refreshRanger());
          }
          return forkJoin(dashlets$).pipe(defaultIfEmpty([]));
        }),
        tap(() => (this.refreshInProgress = false)),
      )
      .subscribe();
```